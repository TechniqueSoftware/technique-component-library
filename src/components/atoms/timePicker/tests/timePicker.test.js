import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import TimePicker from '..'

const fn = jest.fn()
const props = {
  onChange: fn
}

describe('<TimePicker>', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<TimePicker {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders', () => {
    render(<TimePicker {...props} />)
  })

  it('selects time options', async () => {
    render(<TimePicker {...props} />)
    const hourSelected = { value: '2', name: '2' }
    const minuteSelected = { value: '15', name: '15' }
    const meridiemSelected = { value: 'PM', name: 'PM' }

    // Select hour
    const hourSelect = screen.queryAllByRole('combobox')[0]
    await userEvent.selectOptions(hourSelect, [hourSelected.value])
    expect(props.onChange).toHaveBeenCalledWith(hourSelected.value, undefined, 'AM')

    // Select minute
    const minuteSelect = screen.queryAllByRole('combobox')[1]
    await userEvent.selectOptions(minuteSelect, [minuteSelected.value])
    expect(props.onChange).toHaveBeenCalledWith(hourSelected.value, minuteSelected.value, 'AM')

    // Select meridiem
    const meridiemButton = screen.queryAllByRole('radio')[1]
    userEvent.click(meridiemButton)
    expect(props.onChange).toHaveBeenCalledWith(
      hourSelected.value,
      minuteSelected.value,
      meridiemSelected.value)
  })
})
