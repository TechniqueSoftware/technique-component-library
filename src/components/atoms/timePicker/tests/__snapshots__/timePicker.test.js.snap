// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimePicker> should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <select
      class="timepicker__select form-select"
      role="combobox"
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
      <option
        value="4"
      >
        4
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="6"
      >
        6
      </option>
      <option
        value="7"
      >
        7
      </option>
      <option
        value="8"
      >
        8
      </option>
      <option
        value="9"
      >
        9
      </option>
      <option
        value="10"
      >
        10
      </option>
      <option
        value="11"
      >
        11
      </option>
      <option
        value="12"
      >
        12
      </option>
    </select>
    :
    <select
      class="timepicker__select form-select"
    >
      <option
        value="00"
      >
        00
      </option>
      <option
        value="15"
      >
        15
      </option>
      <option
        value="30"
      >
        30
      </option>
      <option
        value="45"
      >
        45
      </option>
    </select>
    <div
      class=" btngroup btn-group"
      role="group"
    >
      <input
        autocomplete="off"
        checked=""
        class="btn-check"
        id="radio-AM"
        name="am-pm"
        type="radio"
        value="AM"
      />
      <label
        class="btn btn-outline-primary"
        for="radio-AM"
        tabindex="0"
      >
        AM
      </label>
      <input
        autocomplete="off"
        class="btn-check"
        id="radio-PM"
        name="am-pm"
        type="radio"
        value="PM"
      />
      <label
        class="btn btn-outline-primary"
        for="radio-PM"
        tabindex="0"
      >
        PM
      </label>
    </div>
  </div>
</DocumentFragment>
`;
